misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/BinaryExpressionEvaluator.java

```diff
@@ -387,6 +403,50 @@ private BExpressionValue conditionalReturn(BVariable lVar, BVariable rVar) {
         }
     }
 
+    /**
+     * Checks for deep value equality.
+     */
+    private BExpressionValue checkValueEquality(BVariable lVar, BVariable rVar, SyntaxKind operatorType)
+            throws EvaluationException {
+        List<Value> argList = new ArrayList<>();
+        argList.add(getValueAsObject(context, lVar));
+        argList.add(getValueAsObject(context, rVar));
+
+        List<String> argTypeNames = new ArrayList<>();
+        argTypeNames.add(JAVA_OBJECT_CLASS);
+        argTypeNames.add(JAVA_OBJECT_CLASS);
```

Comment:
Seems like you are adding the `JAVA_OBJECT_CLASS` twice. This is redundant.

Reply:
This is intentional. This relevant method accepts two `java.lang.Object` instances as its arguments.